scripts/main.js

```diff
@@ -308,9 +314,89 @@ window.addEventListener('WebChannelMessageToContent', function (event) {
       selectNextItem(-1);
     }
   }
+
+  // collecting session metrics on open and closed message types
+  if (message.type === 'popupopen') {
+
+    sessionMetrics.set({startTime: new Date().getTime()});
+
+  } else if (message.type === 'popupclose') {
+
+    sessionMetrics.set({endTime: new Date().getTime()}); 
+    sessionMetrics.set({duration:sessionMetrics.getDuration()});
+
+    if (sessionMetrics.attributes.hasEngaged) {
+
+      var selectedGroup = $('li.selected').parent().parent();
+      var resultIndex = selectedGroup.find('.selected').index();
+
+      sessionMetrics.set({resultType: selectedGroup.attr('data-region')});
+      sessionMetrics.set({resultIndex: resultIndex});
+
+    }
+
+    console.log(sessionMetrics.attributes);
+
+  }
+
 });
 
 // Mouse events should clear arrow selection for now
 $(window).mouseenter(function () {
   $('li.selected').removeClass('selected');
 });
+
+
+$(window).mousemove(function () {
```

Comment:
A cleaner way to hook into existing behavior would be for the metrics object to subscribe to a JS event fired by the view code when a click event or enter key event is detected.